pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+released as a single tested, coherent package with a defined Major, Minor,
+Revision number.  The mof for these files is packaged as a single zip
+file and is available on the DMTF web site.
+
+The DMTF schema is the core set of mofs that represent the models used in
+most CIM/WBEM implementations that that utilize pywbem.
+
+The DMTFSchema class installs a DMTF MOF schema into a directory and makes
+information on that directory availableas a property. Donwloads a DMTF schema
+defined by its version number from the DMTF web site and expands it into a
+directory defined by a constructor input parameter.
+
+The version defined will be downloaded in the directory defined by the
+'schema_dir' constructor parameter if that directory is empty does not exist
+and expanded from the downloaded zip file if the expanded files do not exist.
+
+Installation into the 'schema_dir' directory includes:
+
+1. If necessary, creates the directory defined by schema_dir.
+
+2. Downloads the schema from the DMTF web site into the schema_dir.
+
+3. Expands the zip file for the mof into the directory 'mof<schemaVersionNumber'
+   in the schema_dir.
+
+Multiple DMTF schemas may be maintained in the same 'schema_dir' simultaneously
+because each schema is expanded into a directory identified by the
+schema version information.
+
+Once a DMTF schema is expanded into the individual schema files it consumes
+considerable space since it expands to almost 3000 files.  Therefore it is
+logical to remove all of the individual mof files when not being used and
+also if the schema information is to be committed to be used in testing in
+an environment like github. Therefore, a method :meth:`~pywbem.DMTFSchema.clean`
+removes all of the mof files from local storage.  They are restored the next
+time the :class:`DMTFSchema` object for the same version is constructed.
+
+The DMTF maintains two versions of each released schema:
+
+* Final - This schema does not contain any of the schema classes that are
+  marked experimantel.  It is considered the stable release.
+
+* Experimental - Contains the classes in the 'Final' zip file plus other
+  classes that are considered experimental. Typically it is considered less
+  stable.
+
+:class:`DMTFSchema` will optionally load the experimental schema if the
+constructor parameter 'user-experimental' is 'True'.
+
+The DMTFSchema class also provides a method to create a MOF include file
+from the DMTF Schema downloaded that includes only a subset of the classes
+actually in the repository for compile into a MOF repository.
+"""
+import os
+from zipfile import ZipFile
+import shutil
+import six
+
+if six.PY2:
+    # pylint: disable=wrong-import-order
+    from urllib2 import urlopen
+else:
+    # pylint: disable=wrong-import-order
+    from urllib.request import urlopen
+
+
+__all__ = ['DMTFSchema']
+
+
+class DMTFSchema(object):
+    """
+    This class manages the download and install of these files into a form
```

Comment:
I'd rather describe the class as what an object of the class represents (a downloaded DMTF schema)

Reply:
Added this sentence but kept the old as a second paragraph.

Reply:
DONE